ImageCacheProvider.js

```
@@ -118,27 +119,57 @@ function downloadImage(fromUrl, toFile, headers = {}) {
-                    // cleanup
-                    delete activeDownloads[toFile];
-                });
-        });
+        activeDownloads[toFile] = RNFetchBlob
+            .config({path: tmpFile})
+            .fetch('GET', fromUrl, headers)
+            .then(res => {
+                // console.log(res.respInfo.status, fromUrl)
+                // debug.network(fromUrl);
+                // debug.networkEnd(fromUrl, {
+                //     'General': {
+                //         url: fromUrl,
+                //         method: 'GET',
+                //         status: res.respInfo.status,
+                //         query: null,
+                //     },
+                //     'Response Headers': headers,
+                //     'Response': {
+                //         Raw: toFile
+                //     }
+                // });
+                if (res.respInfo.status === 304) {
+                    return Promise.resolve(toFile);
+                }
+                let status = Math.floor(res.respInfo.status / 100);
+                if (status !== 2) {
+                    throw new Error(res.respInfo.status + ' Failed to successfully download image');
+                }
+                // The download is complete and rename the temporary file
+                setFileInfo(toFile, headers);
+                return fs.mv(tmpFile, toFile);
+            })
+            .catch(err => {
+                // console.log(err.code, fromUrl)
+                // debug.network(fromUrl);
+                // debug.networkEnd(fromUrl, {
+                //     'General': {
+                //         url: fromUrl,
+                //         method: 'GET',
+                //         status: err.code,
+                //         query: null,
+                //     },
+                //     'Response Headers': headers,
+                //     'Response': {
+                //         Error: err
+                //     }
+                // });
+            })
```

Same with this catch that will swallow all errors, unless that is the intention?